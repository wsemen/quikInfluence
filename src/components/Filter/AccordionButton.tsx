import { MinusIcon, AddIcon } from '@chakra-ui/icons';
import { AccordionButton, Box } from '@chakra-ui/react';

const CustomAccordionButton = ({
  label,
  isExpanded,
}: {
  label: string;
  isExpanded: boolean;
}) => {
  return (
    <h2>
      <AccordionButton pl="unset" py={6}>
        <Box
          fontSize="12px"
          flex="1"
          textAlign="left"
          textTransform="capitalize"
        >
          {label}
        </Box>
        {isExpanded ? (
          <MinusIcon fontSize="12px" />
        ) : (
          <AddIcon fontSize="12px" />
        )}
      </AccordionButton>
    </h2>
  );
};

export default CustomAccordionButton;
