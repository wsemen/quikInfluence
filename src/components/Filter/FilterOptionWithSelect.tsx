import {
  AccordionPanel,
  Flex,
  Tag,
  TagCloseButton,
} from '@chakra-ui/react';
import DropdownSelect from 'components/DropdownSelect';
import React, { useEffect, useState } from 'react';
import { FilterItemProps } from 'types';
import CustomAccordionButton from './AccordionButton';

const FilterOptionWithSelect = ({ isExpanded, onChange, title = '', selectOptions }: FilterItemProps) => {
  const [selectedOpt, setSelectedAge] = useState<Array<string>>([]);

  const handleChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    if (!selectedOpt.includes(e.target.value)) {
      setSelectedAge([...selectedOpt, e.target.value]);
    }
  };

  const handleRemoveOpt = (removeOpt: string) => {
    setSelectedAge(selectedOpt.filter(opt => opt !== removeOpt));
  };

  useEffect(() => {
    onChange({ [title]: selectedOpt });
  }, [selectedOpt]);

  return (
    <>
      <CustomAccordionButton label={title} isExpanded={isExpanded} />
      <AccordionPanel pl="unset" pb={4}>
        <DropdownSelect
          onChange={handleChange}
          options={selectOptions || []}
        />
        <Flex flexWrap="wrap" spacing={4} mt={5}>
          {selectedOpt &&
            selectedOpt.map((opt, index) => (
              <Tag
                size="lg"
                key={`${opt}_${index}`}
                variant="solid"
                borderRadius="full"
                colorScheme="blackAlpha"
              >
                {opt}
                <TagCloseButton data-test-id="remove-opt" onClick={() => handleRemoveOpt(opt)} />
              </Tag>
            ))}
        </Flex>
      </AccordionPanel>
    </>
  );
};

export default FilterOptionWithSelect;
