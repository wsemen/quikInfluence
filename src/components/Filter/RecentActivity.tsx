import { AccordionPanel, HStack, Tag } from '@chakra-ui/react';
import { FilterItemProps } from 'types';
import CustomAccordionButton from './AccordionButton';

const RecentActivity = ({ isExpanded }: FilterItemProps) => (
  <>
    <CustomAccordionButton label="Recent Activity" isExpanded={isExpanded} />
    <AccordionPanel pl="unset" pb={4}>
      <HStack spacing={4}>
        {['Email Open', 'Sms Open', 'Non tag'].map(tag => (
          <Tag
            size="lg"
            key={tag}
            variant="solid"
            borderRadius="full"
            colorScheme="blackAlpha"
          >
            {tag}
          </Tag>
        ))}
      </HStack>
    </AccordionPanel>
  </>
);

export default RecentActivity;
