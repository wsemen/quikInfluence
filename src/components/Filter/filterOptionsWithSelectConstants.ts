export default {
  status: [
    { label: "Active", value: "active" },
    { label: "Inactive", value: "inactive" },
    { label: "Incomplete", value: "incomplete" },
    { label: "Complete", value: "complete" }
  ],
  age: [
    { label: "18-25", value: "18-25" },
    { label: "26-32", value: "26-32" },
    { label: "33-50", value: "33-50" },
    { label: "51-above", value: "51-above" }
  ],
  gender: [
    { label: "Male", value: "male" },
    { label: "Female", value: "female" },
    { label: "Other", value: "other" }
  ],
  application: [
    { label: "QuikSession", value: "quikSession" },
    { label: "IdemandBeauti", value: "idemandBeauti" },
    { label: "ASD Exchange", value: "asd_exchange" }
  ]
};
