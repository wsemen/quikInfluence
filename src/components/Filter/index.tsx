import {
  Accordion,
  AccordionItem,
  Heading,
  Stack,
  useColorMode,
} from '@chakra-ui/react';
import queryString from 'query-string';
import CustomButton from 'components/Button';
import React from 'react';
import { T } from 'types';
import { cardThemeColor } from 'utils/constants/colorConstants';
import FilterOptionWithSelect from './FilterOptionWithSelect';
import RecentActivity from './RecentActivity';
import { useRouter } from 'next/router';
import filterOptionsWithSelectConstants from './filterOptionsWithSelectConstants';

const Filter = () => {
  const router = useRouter() || { push: () => {} };
  const { colorMode } = useColorMode();
  const [filter, setFilter] = React.useState<T>({});

  const handleChange = (val: any) => {
    setFilter({ ...filter, ...val });
  };

  const handleClick = () => {
    const _queryString = queryString.stringify(filter, {
      arrayFormat: 'comma',
    });
    console.log(_queryString); // Result: age=26-32,18-25,33-50&gender=female,male
    router.push(`?${_queryString}`);
  };

  return (
    <Stack p={10} bg={cardThemeColor[colorMode]}>
      <Heading size="md">Filter</Heading>
      <Accordion mt={5} allowMultiple>
        <AccordionItem border="none">
          {({ isExpanded }) => (
            <RecentActivity onChange={handleChange} isExpanded={isExpanded} />
          )}
        </AccordionItem>

        {Object.keys(filterOptionsWithSelectConstants).map(
          (filterItem, index) => (
            <AccordionItem key={index} border="none">
              {({ isExpanded }) => (
                <FilterOptionWithSelect
                  onChange={handleChange}
                  isExpanded={isExpanded}
                  title={filterItem}
                  // @ts-ignore
                  selectOptions={filterOptionsWithSelectConstants[filterItem]}
                />
              )}
            </AccordionItem>
          )
        )}
      </Accordion>
      <CustomButton onClick={handleClick} variant="outline">
        Filter Search
      </CustomButton>
    </Stack>
  );
};

export default Filter;
