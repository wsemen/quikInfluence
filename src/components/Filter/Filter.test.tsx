import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Filter from '.';

it('renders Filter component', () => {
  const { container, getByText } = render(<Filter />);

  userEvent.click(getByText('Filter Search'));
  expect(container).toMatchSnapshot();
});
