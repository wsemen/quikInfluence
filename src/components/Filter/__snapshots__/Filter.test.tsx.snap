// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Filter component 1`] = `
<div>
  <div
    class="chakra-stack css-17r4hs2"
  >
    <h2
      class="chakra-heading css-0"
    >
      Filter
    </h2>
    <div
      class="chakra-accordion css-odz94x"
    >
      <div
        class="chakra-accordion__item css-cn9ye4"
      >
        <h2>
          <button
            aria-controls="accordion-panel-1"
            aria-expanded="false"
            class="chakra-accordion__button css-12ttzao"
            data-index="0"
            id="accordion-button-1"
            type="button"
          >
            <div
              class="css-8tcdg7"
            >
              Recent Activity
            </div>
            <svg
              class="chakra-icon css-1hiix9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-1"
            class="chakra-accordion__panel css-4rbilf"
            id="accordion-panel-1"
            role="region"
          >
            <div
              class="chakra-stack css-1qwdsol"
            >
              <span
                class="css-1ehzxp"
              >
                Email Open
              </span>
              <span
                class="css-1ehzxp"
              >
                Sms Open
              </span>
              <span
                class="css-1ehzxp"
              >
                Non tag
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="chakra-accordion__item css-cn9ye4"
      >
        <h2>
          <button
            aria-controls="accordion-panel-3"
            aria-expanded="false"
            class="chakra-accordion__button css-12ttzao"
            data-index="1"
            id="accordion-button-3"
            type="button"
          >
            <div
              class="css-8tcdg7"
            >
              status
            </div>
            <svg
              class="chakra-icon css-1hiix9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-3"
            class="chakra-accordion__panel css-4rbilf"
            id="accordion-panel-3"
            role="region"
          >
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <div
                class="chakra-select__wrapper css-42b2qy"
              >
                <select
                  class="chakra-select css-l65hh7"
                  data-test-id="select-component"
                  id=""
                >
                  <option
                    value=""
                  >
                    Select ---
                  </option>
                  <option
                    data-testid="select-option"
                    value="active"
                  >
                    Active
                  </option>
                  <option
                    data-testid="select-option"
                    value="inactive"
                  >
                    Inactive
                  </option>
                  <option
                    data-testid="select-option"
                    value="incomplete"
                  >
                    Incomplete
                  </option>
                  <option
                    data-testid="select-option"
                    value="complete"
                  >
                    Complete
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="css-bkneeg"
              spacing="4"
            />
          </div>
        </div>
      </div>
      <div
        class="chakra-accordion__item css-cn9ye4"
      >
        <h2>
          <button
            aria-controls="accordion-panel-5"
            aria-expanded="false"
            class="chakra-accordion__button css-12ttzao"
            data-index="2"
            id="accordion-button-5"
            type="button"
          >
            <div
              class="css-8tcdg7"
            >
              age
            </div>
            <svg
              class="chakra-icon css-1hiix9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-5"
            class="chakra-accordion__panel css-4rbilf"
            id="accordion-panel-5"
            role="region"
          >
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <div
                class="chakra-select__wrapper css-42b2qy"
              >
                <select
                  class="chakra-select css-l65hh7"
                  data-test-id="select-component"
                  id=""
                >
                  <option
                    value=""
                  >
                    Select ---
                  </option>
                  <option
                    data-testid="select-option"
                    value="18-25"
                  >
                    18-25
                  </option>
                  <option
                    data-testid="select-option"
                    value="26-32"
                  >
                    26-32
                  </option>
                  <option
                    data-testid="select-option"
                    value="33-50"
                  >
                    33-50
                  </option>
                  <option
                    data-testid="select-option"
                    value="51-above"
                  >
                    51-above
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="css-bkneeg"
              spacing="4"
            />
          </div>
        </div>
      </div>
      <div
        class="chakra-accordion__item css-cn9ye4"
      >
        <h2>
          <button
            aria-controls="accordion-panel-7"
            aria-expanded="false"
            class="chakra-accordion__button css-12ttzao"
            data-index="3"
            id="accordion-button-7"
            type="button"
          >
            <div
              class="css-8tcdg7"
            >
              gender
            </div>
            <svg
              class="chakra-icon css-1hiix9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-7"
            class="chakra-accordion__panel css-4rbilf"
            id="accordion-panel-7"
            role="region"
          >
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <div
                class="chakra-select__wrapper css-42b2qy"
              >
                <select
                  class="chakra-select css-l65hh7"
                  data-test-id="select-component"
                  id=""
                >
                  <option
                    value=""
                  >
                    Select ---
                  </option>
                  <option
                    data-testid="select-option"
                    value="male"
                  >
                    Male
                  </option>
                  <option
                    data-testid="select-option"
                    value="female"
                  >
                    Female
                  </option>
                  <option
                    data-testid="select-option"
                    value="other"
                  >
                    Other
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="css-bkneeg"
              spacing="4"
            />
          </div>
        </div>
      </div>
      <div
        class="chakra-accordion__item css-cn9ye4"
      >
        <h2>
          <button
            aria-controls="accordion-panel-9"
            aria-expanded="false"
            class="chakra-accordion__button css-12ttzao"
            data-index="4"
            id="accordion-button-9"
            type="button"
          >
            <div
              class="css-8tcdg7"
            >
              application
            </div>
            <svg
              class="chakra-icon css-1hiix9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                fill="currentColor"
              />
            </svg>
          </button>
        </h2>
        <div
          class="chakra-collapse"
          style="overflow: hidden; display: none; opacity: 0; height: 0px;"
        >
          <div
            aria-labelledby="accordion-button-9"
            class="chakra-accordion__panel css-4rbilf"
            id="accordion-panel-9"
            role="region"
          >
            <div
              class="chakra-form-control css-0"
              role="group"
            >
              <div
                class="chakra-select__wrapper css-42b2qy"
              >
                <select
                  class="chakra-select css-l65hh7"
                  data-test-id="select-component"
                  id=""
                >
                  <option
                    value=""
                  >
                    Select ---
                  </option>
                  <option
                    data-testid="select-option"
                    value="quikSession"
                  >
                    QuikSession
                  </option>
                  <option
                    data-testid="select-option"
                    value="idemandBeauti"
                  >
                    IdemandBeauti
                  </option>
                  <option
                    data-testid="select-option"
                    value="asd_exchange"
                  >
                    ASD Exchange
                  </option>
                </select>
                <div
                  class="chakra-select__icon-wrapper css-162mkon"
                >
                  <svg
                    aria-hidden="true"
                    class="chakra-select__icon"
                    focusable="false"
                    role="presentation"
                    style="width: 1em; height: 1em; color: currentColor;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                      fill="currentColor"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div
              class="css-bkneeg"
              spacing="4"
            />
          </div>
        </div>
      </div>
    </div>
    <button
      class="chakra-button css-1i4yx9k"
      type="button"
    >
      Filter Search
    </button>
  </div>
</div>
`;
